### PR TITLE
Migrate to browser polyfill for ui-wrapper.

### PR DESCRIPTION
**Pull request by jonathanKingston**
Additionally support debugging tabId param for the panel.

Ignore session errors for browser.storage.session as not in the types.

**Reviewer:** @shakyShane

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/ui/base/ui-wrapper.es6.js [hunk 1]
```diff
@@ -9,54 +9,67 @@ const sendMessage = async (messageType, options) => {
...
-    })
+/** @typedef {ReturnType<import('../../background/message-handlers.js').getTab>} Tab */
+
+/**
+ * @returns {Promise<Tab|undefined>}
+ */
+async function getBackgroundTabData () {
+    const url = new URL(window.location.href)
+    let tabId
+    // Used for debugging only
```
Comment (kzar) [inline shared/js/ui/base/ui-wrapper.es6.js:29]: Would you mind to expand this comment a little bit, to explain how it's used for debugging? E.g. opening popup UI in a new tab and then setting the tab ID in URL manually.

File: shared/js/ui/models/site-company-list.es6.js [hunk 2]
```diff
@@ -17,18 +18,14 @@ SiteCompanyList.prototype = window.$.extend({},
 
         fetchAsyncData: function () {
             return new Promise((resolve, reject) => {
-                this.sendMessage('getCurrentTab').then((tab) => {
-                    if (tab) {
```
Comment (kzar) [inline shared/js/ui/models/site-company-list.es6.js:21]: Is this check not necessary any more?

File: shared/js/ui/models/site-company-list.es6.js [hunk 3]
```diff
@@ -17,18 +18,14 @@ SiteCompanyList.prototype = window.$.extend({},
 
         fetchAsyncData: function () {
```
Comment (kzar) [inline shared/js/ui/models/site-company-list.es6.js:19]: Optional: Would be nice to tidy this function up a little, e.g. using async/await, the new [method definitions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions) syntax and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). Maybe out of scope though.

File: shared/js/ui/models/site-company-list.es6.js (hunk 2 above)
Comment (jonathanKingston) [reply shared/js/ui/models/site-company-list.es6.js:21]: er maybe for _updateCompaniesList let me double check. Otherwise it's functionally equiv.

Comment (jonathanKingston) [reply shared/js/ui/models/site-company-list.es6.js:21]: For paranoia sake, I added it back in. Pretty sure it's not needed.

File: shared/js/ui/models/site-company-list.es6.js (hunk 3 above)
Comment (jonathanKingston) [reply shared/js/ui/models/site-company-list.es6.js:19]: The whole file is going in the bin soon anyway tbh. Let's limit diff noise but I take your point I've been making these kinda changes where possible.

Comment (kzar) [reply shared/js/ui/models/site-company-list.es6.js:19]: OK fair enough.

**Review by kzar — APPROVED** [review]

**Comment by kzar** [comment]
LGTM, I'm obviously not too familiar with this code though, so up to you if you want to wait for @shakyShane or someone to also take a pass.

**Comment by jonathanKingston** [comment]
> LGTM, I'm obviously not too familiar with this code though, so up to you if you want to wait for @shakyShane or someone to also take a pass.

Cheers, I'll let him know but I'll merge for now as discussed over MM.

**Pull request merged**